deploy_tools/compile.py

```diff
@@ -0,0 +1,151 @@
+    result: Dict[str, Dict] = {}
+
+    for source_path, file_contracts in compiled_contracts.items():
+        if source_path not in file_paths:
+            continue
+
+        for contract_name, raw_contract_data in file_contracts.items():
+            contract_data = normalize_contract_data(raw_contract_data)
+            if contract_name not in result:
+                result[contract_name] = contract_data
+            else:
+                raise BaseException('Can not compile two contracts with the same name')
+
+    return result
+
+
+def log_compilation_errors(errors: List[Dict]):
+    for error in errors:
+        if 'formattedMessage' in error:
+            print(error['formattedMessage'])
+        else:
+            print(error['message'])
+
+
+def compile_project(
+        contracts_path: str = None,
+        *,
+        file_paths: List[str] = None,
+        allow_paths: List[str] = None,
+        pattern='*.sol'
+):
+    """
+    Compiles all contracts of the project into a single output
+    Args:
+        contracts_path: The path of the folder that includes the contracts
+        file_paths: Optional a list of to compiled contracts can be provided
+        allow_paths: Additional paths from where it is allowed to load contracts
+        pattern: The pattern to find the solidity files
+
+    Returns: A dictionary containing the compiled assets of the contracts
+
+    """
+
+    if file_paths is None:
+        file_paths = []
+
+    if allow_paths is None:
+        allow_paths = []
+
+    if contracts_path is None and not file_paths:
+        contracts_path = 'contracts'
+
+    if contracts_path is not None:
+        file_paths.extend(find_files(contracts_path, pattern=pattern))
+        allow_paths.append(contracts_path)
+
+    sources = load_sources(file_paths)
+
+    std_input = {
+        'language': 'Solidity',
+        'sources': sources,
+        'settings': {
+            'outputSelection': {
+                '*': {
+                    '*': DEFAULT_OUTPUT_SELECTION
+                }
+            }
+        }
+    }
+
+    compilation_result = compile_standard(
+        std_input,
+        allow_paths=','.join(os.path.abspath(path) for path in allow_paths)
+    )
+
+    if 'errors' in compilation_result:
+        log_compilation_errors(compilation_result['errors'])
+
+    compiled_contracts = normalize_compiled_contracts(compilation_result['contracts'], file_paths)
+    return compiled_contracts
+
+
+def compile_contract(
+        name: str,
+        *,
+        contracts_path='contracts',
+        file_extension='.sol'
+):
+    filename = name + file_extension
+    file_paths = list(find_files(contracts_path, filename))
+
+    if len(file_paths) < 1:
+        raise ValueError('File not found: {}'.format(filename))
+
+    if len(file_paths) > 1:
+        raise ValueError('Multiple files found: {}'.format(file_paths))
+
+    compiled_contracts = compile_project(file_paths=file_paths, allow_paths=[contracts_path])
```

Comment:
Not entirely sure what the goal of this function is, but why do have only the path to contracts as `allow_paths` and not an argument of this function?

Reply:
I am really not sure if we will need this function so I would just keep it for now like this, and extend it if we need it.